src/NuGet.Core/NuGet.ProjectModel/LockFile/LockFile.cs

```diff
@@ -128,7 +128,45 @@ public bool Equals(LockFile other)
                 && Libraries.OrderedEquals(other.Libraries, library => library.Name, StringComparer.OrdinalIgnoreCase)
                 && Targets.OrderedEquals(other.Targets, target => target.Name, StringComparer.Ordinal)
                 && PackageFolders.SequenceEqual(other.PackageFolders)
-                && EqualityUtility.EqualsWithNullCheck(PackageSpec, other.PackageSpec);
+                && EqualityUtility.EqualsWithNullCheck(PackageSpec, other.PackageSpec)
+                && LogsEqual(other.LogMessages);
+        }
+
+        private bool LogsEqual(IList<IAssetsLogMessage> otherLogMessages)
```

Comment:
Can we use [`Enumerable.SequenceEqual`](https://docs.microsoft.com/en-us/dotnet/api/system.linq.enumerable.sequenceequal?view=netframework-4.8) instead of custom implementation to compare the log messages ?